pywbem/_subscription_manager.py

```diff
@@ -870,9 +886,12 @@ def add_filter(self, server_id, source_namespace, query,
             raise ValueError("The name parameter cannot be used to add "
                              "owned filters.")
 
-        filter_inst = self._create_filter(server_id, source_namespace, query,
-                                          query_language, owned, filter_id,
-                                          name)
+        if isinstance(source_namespaces, six.string_types):
```

Comment:
That is a dangerous check. What if a byte string is passed in? I suggest:
`if not isinstance(source_namespaces, Sequence):`
where Sequence first gets imported from collections.abc and upon importerror from collections.

Reply:
I am confused.  My understanding is that `sequence` may be any of at least list, tuple, string, and range;    I believe that you get down into mutable and immutable sequences before resolving this one.

string - immutable sequence
list - mutable sequence
tuple - immutable sequence

Perhaps the better solution here is to validate that the value is either str or mutable sequence (list) and reject if not. Then we know exactly  what we Have.  I impemented that.

Since we have six.string_types tests in many places to determine if we have a list or string, do we have a much bigger problem here????

Reply:
You are right. Strings are also sequences, and even worse, I misread the current test, which uses `six.string_types` and thus covers both unicode strings and byte strings. Sorry for my confusion on this one.
WITHDRAWING COMMENT

Reply:
OK.  I spent another hour going back over the whole byte string, str, sequence, etc.  and to be honest, it is really messy.

I went ahead and did check on string_types or list for validation of the parameter in add_filter.  Thus, no byte strings or tuples.